Use xcconfig files

Odrakir opened the pull request:
## Info
This PR adds xcconfig files for every target in the app (main app, frameworks and tests). Together with some other base xcconfig files to share settings.

The idea is we stop making changes to build settings through the Xcode plist editor and instead use these files which are much easier to understand and less prone to merge conflicts.

### Technical considerations
Every target build configuration file inherits from a base file and also optionally (with "#include?") from a CocoaPods file. That's needed for CocoaPods to work and if it's missing it'll let us know when running "pod install"

There's still some inheritance that should be done. If you see some low hanging fruits, go ahead and create an issue. I think for this PR this is enough.

Next thing we should do is creating a static framework project template that generates those config files for us too.

### Relationships
Closes #128 

## Repository health

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
<!-- Check if used so that we can track template usage -->
- [ ] The code used custom templates

## Reviewing

### Pointers

Commits start very granular and easy to understand, and get bigger over time. Just follow the yellow brick road.

### Review template

<!-- General -->
- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones
<!-- UI related -->
- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Let me merge it

sofokli1 commented on Modules/Analytics/Configurations/Project-Shared.xcconfig:
```diff
@@ -0,0 +1,5 @@
+
+CURRENT_PROJECT_VERSION = 1
+VERSION_INFO_PREFIX =
+VERSIONING_SYSTEM = apple-generic
```
These are duplicated in many all modules

sofokli1 commented on Modules/Architecture/Configurations/Architecture-Debug.xcconfig:
```diff
@@ -0,0 +1,4 @@
+#include "Architecture-Shared.xcconfig"
+#include? "../../Pods/Target Support Files/Pods-Architecture/Pods-Architecture.debug.xcconfig"
+
+SWIFT_OPTIMIZATION_LEVEL = -Onone
```
this is duplicated in all modules

sofokli1 commented on Modules/Architecture/Configurations/Project-Debug.xcconfig:
```diff
@@ -0,0 +1,11 @@
+MTL_ENABLE_DEBUG_INFO = INCLUDE_SOURCE
+ONLY_ACTIVE_ARCH = YES
+SWIFT_ACTIVE_COMPILATION_CONDITIONS = DEBUG
+SWIFT_OPTIMIZATION_LEVEL = -Onone
```
same here

sofokli1 commented on Modules/Architecture/Configurations/Project-Release.xcconfig:
```diff
@@ -0,0 +1,8 @@
+MTL_ENABLE_DEBUG_INFO = NO
+SWIFT_COMPILATION_MODE = wholemodule
+SWIFT_OPTIMIZATION_LEVEL = -O
+VALIDATE_PRODUCT = YES
```
and here

Odrakir commented on Modules/Assets/Configurations/Assets-Shared.xcconfig:
```diff
@@ -0,0 +1,18 @@
+LD_RUNPATH_SEARCH_PATHS = $(inherited) @executable_path/Frameworks @loader_path/Frameworks
+PRODUCT_BUNDLE_IDENTIFIER = com.toggl.aurora.Assets
+PRODUCT_NAME = $(TARGET_NAME:c99extidentifier)
+PROVISIONING_PROFILE_SPECIFIER = 
```
This is commenting old code

sofokli1 replied:
There is an up to date example here https://github.com/toggl/ios/pull/174/files#diff-f5b8c9d2ee4be71ecb9ec2161cdea761R4

sofokli1 replied:
basically in all `Project-Shared.xcconfig` in the modules

sofokli1 commented on App/Configurations/AppTests-Debug.xcconfig:
```diff
@@ -0,0 +1,2 @@
+#include "AppTests-Shared.xcconfig"
+#include? "../../Pods/Target Support Files/Pods-App-AppTests/Pods-App-AppTests.debug.xcconfig"
```
One more question: why have pod xcconfig imports as optional?

Odrakir replied:
My idea is to include this in a template for all new modules. Some of them won't be using Cocoapods (or not at the beginning), but when they start using them, that import will be ready. So that's why.

Odrakir commented:
I removed more duplication. I also changed the name of the files to be "Whatever.debug.xcconfig" instead of "Whatever-Debug.xcconfig" which follows the same naming as CocoaPods.

sofokli1 commented on Modules/Timer/Configurations/Timer.debug.xcconfig:
```diff
@@ -0,0 +1,4 @@
+#include "Timer-Shared.xcconfig"
+#include? "../../Pods/Target Support Files/Pods-Timer/Pods-Timer.debug.xcconfig"
+
+SWIFT_OPTIMIZATION_LEVEL = -Onone
```
Is this something that is needed just for the this module for release?

sofokli1 commented on Modules/Onboarding/Configurations/Onboarding.debug.xcconfig:
```diff
@@ -0,0 +1,4 @@
+#include "Onboarding-Shared.xcconfig"
+#include? "../../Pods/Target Support Files/Pods-Onboarding/Pods-Onboarding.debug.xcconfig"
+
+SWIFT_OPTIMIZATION_LEVEL = -Onone
```
Is this something that is needed just for the this module for debug?

Odrakir commented:
About the changes to xcscheme files. Most of them were not configured properly to run tests for some reason. I guess those two were ok.
